Source/Core/DiscIO/VolumeVerifier.cpp

```diff
@@ -0,0 +1,912 @@
+    // The only game that has that many partitions in total is Super Smash Bros. Brawl,
+    // and that game places all partitions other than UPDATE and DATA in the second table.
+    AddProblem(Severity::Low,
+               GetStringT("There are too many partitions in the first partition table."));
+  }
+
+  std::vector<u32> types;
+  for (const Partition& partition : partitions)
+  {
+    const std::optional<u32> type = m_volume.GetPartitionType(partition);
+    if (type)
+      types.emplace_back(*type);
+  }
+
+  if (std::find(types.cbegin(), types.cend(), PARTITION_UPDATE) == types.cend())
+    AddProblem(Severity::Low, GetStringT("The update partition is missing."));
+
+  if (std::find(types.cbegin(), types.cend(), PARTITION_DATA) == types.cend())
+    AddProblem(Severity::High, GetStringT("The data partition is missing."));
+
+  const bool has_channel_partition =
+      std::find(types.cbegin(), types.cend(), PARTITION_CHANNEL) != types.cend();
+  if (ShouldHaveChannelPartition() && !has_channel_partition)
+    AddProblem(Severity::Medium, GetStringT("The channel partition is missing."));
+
+  const bool has_install_partition =
+      std::find(types.cbegin(), types.cend(), PARTITION_INSTALL) != types.cend();
+  if (ShouldHaveInstallPartition() && !has_install_partition)
+    AddProblem(Severity::High, GetStringT("The install partition is missing."));
+
+  if (ShouldHaveMasterpiecePartitions() &&
+      types.cend() ==
+          std::find_if(types.cbegin(), types.cend(), [](u32 type) { return type >= 0xFF; }))
+  {
+    // i18n: This string is referring to a game mode in Super Smash Bros. Brawl called Masterpieces
+    // where you play demos of NES/SNES/N64 games. Official translations:
+    // 名作トライアル (Japanese), Masterpieces (English), Meisterstücke (German), Chefs-d'œuvre
+    // (French), Clásicos (Spanish), Capolavori (Italian), 클래식 게임 체험판 (Korean).
+    // If your language is not one of the languages above, consider leaving the string untranslated
+    // so that people will recognize it as the name of the game mode.
+    AddProblem(Severity::Medium, GetStringT("The Masterpiece partitions are missing."));
+  }
+
+  for (const Partition& partition : partitions)
+  {
+    if (m_volume.GetPartitionType(partition) == PARTITION_UPDATE && partition.offset != 0x50000)
+    {
+      AddProblem(Severity::Low, GetStringT("The update partition is not at its normal position."));
+    }
+
+    const u64 normal_data_offset = m_volume.IsEncryptedAndHashed() ? 0xF800000 : 0x838000;
+    if (m_volume.GetPartitionType(partition) == PARTITION_DATA &&
+        partition.offset != normal_data_offset && !has_channel_partition && !has_install_partition)
+    {
+      AddProblem(
+          Severity::Low,
+          GetStringT("The data partition is not at its normal position. This will affect the "
+                     "emulated loading times. When using netplay or sending input recordings to "
```

Comment:
```suggestion
                     "emulated loading times. When using NetPlay or sending input recordings to "
```

Reply:
NetPlay feels uglier to me, and it's not consistently used throughout Dolphin, but... eh, I guess I can go with it since Dolphin uses NetPlay in more places than netplay.